### PR TITLE
add domain datasource

### PR DESCRIPTION
**Pull request by illi-j**
This PR add domain datasource.

### CODE REVIEW

File: website/docs/d/domain.html.markdown [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+layout: "exoscale"
+page_title: "Exoscale: exoscale_domain"
+sidebar_current: "docs-exoscale-domain"
+description: |-
+  Provides information about a Domain.
+---
+
+# exoscale\_domain
+
+Provides information on an [domain][domain].
```
Comment (falzm) [inline website/docs/d/domain.html.markdown]:
```suggestion
Provides information on a domain name hosted on [Exoscale DNS][exodns].
```

File: website/docs/d/domain.html.markdown [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+[domain]: ../r/domain.html
+
+## Example Usage
+
+```hcl
+data "exoscale_domain" "exo" {
+  name = my-company.com
```
Comment (falzm) [inline website/docs/d/domain.html.markdown]:
```suggestion
  name = "my-company.com"
```

File: website/docs/d/domain.html.markdown [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+# exoscale\_domain
+
+Provides information on an [domain][domain].
+
+[domain]: ../r/domain.html
+
+## Example Usage
+
+```hcl
+data "exoscale_domain" "exo" {
```
Comment (falzm) [inline website/docs/d/domain.html.markdown]:
```suggestion
data "exoscale_domain" "my-company-com" {
```

File: website/docs/d/domain.html.markdown [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+sidebar_current: "docs-exoscale-domain"
+description: |-
+  Provides information about a Domain.
+---
+
+# exoscale\_domain
+
+Provides information on an [domain][domain].
+
+[domain]: ../r/domain.html
```
Comment (falzm) [inline website/docs/d/domain.html.markdown]:
```suggestion
[exodns]: https://www.exoscale.com/dns/
```

File: website/docs/d/domain.html.markdown [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+```hcl
+data "exoscale_domain" "exo" {
+  name = my-company.com
+}
+```
+
+## Argument Reference
+
+* `name` - (Required) The name of the [domain][domain].
```
Comment (falzm) [inline website/docs/d/domain.html.markdown]:
```suggestion
* `name` - (Required) The name of the domain.
```

File: website/docs/d/domain.html.markdown [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+[domain]: https://www.exoscale.com/dns/
```
Comment (falzm) [inline website/docs/d/domain.html.markdown]:
```suggestion
```

File: exoscale/datasource_exoscale_domain_test.go [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+		Steps: []resource.TestStep{
+			{
+				Config: fmt.Sprintf(`
+%s
+data "exoscale_domain" "domain" {
+  name = "${exoscale_domain.exo.name}"
+}`, testAccResourceDomainRecordConfigCreate),
+				Check: resource.ComposeTestCheckFunc(
+					testAccDatasourceDomainAttributes(testAttrs{
+						"name": ValidateString(testDomain),
```
Comment (falzm) [inline exoscale/datasource_exoscale_domain_test.go]: Starting from #35 this won't work anymore: we don't have a global `testDomain` variable with a static domain name, we now generate a random domain name at every acceptance test run.

You should do something in this spirit:

```go
var testAccDatasourceDomainName = testPrefix + "-" + testRandomString() + ".net"

func TestAccDatasourceDomain(t *testing.T) {
	resource.Test(t, resource.TestCase{
		PreCheck:  func() { testAccPreCheck(t) },
		Providers: testAccProviders,
		Steps: []resource.TestStep{
			{
				Config: fmt.Sprintf(`
resource "exoscale_domain" "test" {
  name = "%s"
}

data "exoscale_domain" "test" {
  name = exoscale_domain.test.name
}`, testAccDatasourceDomainName),
				Check: resource.ComposeTestCheckFunc(
					testAccDatasourceDomainAttributes(testAttrs{
						"name": ValidateString(testAccDatasourceDomainName),
					}),
				),
			},
		},
	})
}
```

**Review by falzm — CHANGES_REQUESTED** [review]
A few changes requested, but good work @illi-j: welcome to the very select Terraform provider developers club 😉

File: exoscale/datasource_exoscale_domain.go [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+		return err
+	}
+
+	d.SetId(strconv.FormatInt(domain.ID, 10))
+
+	if err := d.Set("name", domain.Name); err != nil {
+		return err
+	}
+
+	return nil
```
Comment (pierre-emmanuelJ) [inline exoscale/datasource_exoscale_domain.go]:
```suggestion
	return d.Set("name", domain.Name)
```

File: exoscale/datasource_exoscale_domain.go [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+	domain, err := client.GetDomain(ctx, domainName.(string))
+	if err != nil {
+		return err
+	}
+
+	d.SetId(strconv.FormatInt(domain.ID, 10))
+
+	if err := d.Set("name", domain.Name); err != nil {
+		return err
+	}
```
Comment (pierre-emmanuelJ) [inline exoscale/datasource_exoscale_domain.go]: And remove this part ;)

**Review by pierre-emmanuelJ — CHANGES_REQUESTED** [review]

File: exoscale/datasource_exoscale_domain_test.go (hunk 7 above)
Comment (illi-j) [reply exoscale/datasource_exoscale_domain_test.go]: thanks !

**Review by falzm — APPROVED** [review]
🚀

**Review by jessicatoscani — APPROVED** [review]

**Review by pierre-emmanuelJ — APPROVED** [review]
🎁 📀

**Pull request merged**